libraries/stdlib/test/text/StringTest.kt

```diff
@@ -1314,4 +1314,19 @@ ${"    "}
         assertEquals("  ABC\n   \n  123", "ABC\n   \n123".prependIndent("  "))
         assertEquals("  ", "".prependIndent("  "))
     }
+
+    @Test fun charArrayToStringFullSlice() {
+        val chars: CharArray = charArrayOf().plus("Kotlin".toList())
```

Comment:
I wonder if it was tested on strings with non-ASCII characters and surrogate pairs.

Reply:
@sandwwraith well String.fromCharCode does not support surrogate pairs, what can be done then?

Reply:
From https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/fromCharCode#Getting_it_to_work_with_higher_values : 

> Although most common Unicode values can be represented with one 16-bit number (as expected early on during JavaScript standardization) and fromCharCode() can be used to return a single character for the most common values (i.e., UCS-2 values which are the subset of UTF-16 with the most common characters), in order to deal with ALL legal Unicode values (up to 21 bits), fromCharCode() alone is inadequate. Since the higher code point characters use two (lower value) "surrogate" numbers to form a single character, String.fromCodePoint() (part of the ES2015 standard) can be used to return such a pair and thus adequately represent these higher valued characters.

So I suppose you should use `fromCodePoint` instead. In Java, as I remember, char code represents also a whole complete char, while code point can be split to pair

Reply:
However, I just looked up `java.lang.String` documentation and found that it has [separate special constructor](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html#String-int:A-int-int-) for working with code points array. To preserve this semantic in multiplatform code, it's better to leave things without surrogate pairs support. However, we probably need special function for handling code points. @ilya-g what do you think?

Reply:
Given that both `String`  and `CharArray` are sequence of UTF-16 code units, I don't think any special support for surrogate pairs is required in these two functions.
The function that converts an array of Unicode code points to a `String` is out of scope of this PR.

Reply:
OK, than it's better to revert last commit and stick with previously tested version. Sorry for inconvenience.